### DISCUSSION TITLE
3D markov chain for categorical simulations

### DISCUSSION THREAD

**Discussion by RyanConway91** [GeoStat-Framework/GSTools#349] [Q&A]
I am working on county wide models of aquifer classification. I want to simulate the aquifer classification in 3D using transitional probability/markov chains. 

I was planning on using the software T-PROGS (http://gmsdocs.aquaveo.com/t-progs.pdf) but it seems the fitting of the transiograms is limited to 100,000 data points (not mentioned in the documentation!), but I have about 1,000,000 data points per county. 

Is this something GSTools can do? Browsing tutorials/methods I did not see anything that sounded like what I am looking for, but I am new to the terminology, so wanted to ask. 

If not GSTools, is anyone aware of another open source software package that can do this?

**Comment by LSchueler** [comment]
Hi,

in GSTools you can use field transformations, as shown in [this example](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/07_transformations/02_discrete.html#sphx-glr-examples-07-transformations-02-discrete-py). The discrete values of this particular transformation can act as classification indices. And then you want to create interpolated boundaries between the classifications from your large data set? Maybe you could look into tessellation algorithms?

Does that help?

Cheers,
Lennart

**Reply by RyanConway91** [reply]
I am looking specifically for other tools for transition probability modeling using markov chains, not a alternate categorical modeling approach. Thanks for the idea though!

**Comment by jacobgrasmick** [comment]
You could look at the R package spMC. Have used it for quite some time.
I’ve yet to find an equivalent Python package though… let me know if you do!
Jacob

**Comment by RyanConway91** [comment]
SpMC sounds like what I'm looking for.  Thanks!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
